src/iter/mod.rs

```diff
@@ -266,6 +266,23 @@ pub trait ParallelIterator: Sized + Send {
     /// The `init` value will be cloned only as needed to be paired with
     /// the group of items in each rayon job.  It does not require the type
     /// to be `Sync`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::sync::mpsc::channel;
+    /// use rayon::prelude::*;
+    ///
+    /// let (sender, receiver) = channel();
+    ///
+    /// let mut par_iter = (0..5).into_par_iter().map_with(sender, |s, x| { s.send(x).unwrap(); x });
```

Comment:
Nit: can we format this a bit more clearly? Maybe run the example through rustfmt (on play) and paste it back? I feel like it's a bit hard to follow what's happening. (Also, some comments explaining what's going on wouldn't hurt.)